### ISSUE TITLE
cppvsdbg Debugging becomes no-op between 1.4.1 and 1.5.0

### ISSUE THREAD

**Issue by hostilefork** [microsoft/vscode-cpptools#7808]
Bug type: Debugger

- OS and Version: Windows 10
- VS Code Version: 1.58.0
- C/C++ Extension Version: 1.5.0

After an automated upgrade to the C++ extension today from 1.4.1 to 1.5.0, if I request to run with debugging shows no output in any of the logs (Terminal, Debug Console, Output).

I added  `"logging": {"engineLogging": true}` to launch.json, but this provided no output.

I tried downgrading from 1.5.0 to 1.4.1.  That made it work.  Also, in 1.4.1 the debug console honored the engineLogging switch I'd added and has a lot of output (along the lines of `<- (E) {"seq":3,"type":"event","event":"output","body"`).  I'll reiterate that no such output was produced after the 1.5.0 upgrade.

From the console, I tried running the file at:

    %userprofile%\.vscode\extensions\ms-vscode.cpptools-1.5.0\debugAdapters\vsdbg\bin\vsdbg.exe

When I do this in 1.5.0 I get **"FATAL ERROR: Failed to initialize dispatcher with error 8ede0007"**.  

Running the vsdbg.exe from the downgraded ms-vscode.cpptools-1.4.1 directory, I do not get an error.  It just sits there, which is presumably the right thing for it to do.

So I'd guess that FATAL ERROR probably has something to do with the problem.

**Comment by zeroxia** [comment]
Today cpptools prompts an update, and after it, pressing F5 just triggers cmake building, after building success, nothing happens.

(I use cpptools, clangd and cmake-tools)

And I can confirm it's the same as OP, running `vsdbg.exe` returns immediately error:

```
FATAL ERROR: Failed to initialize dispatcher with error 8ede0007
```

I found another computer with cpptools 1.4.1, running `vsdbg.exe` directly will not prompt any error, just stay there.

**Comment by hostilefork** [comment]
I actually discovered there seems to be an official method to go back on the extensions.  Someone should update the page I found as the top hit for "downgrade visual studio extensions"!!

You can simply click the settings gear on the extension in the list and choose **"Install Another Version"**.  The UI lets you keep it or update it without fiddling configs.

**Comment by XNephila** [comment]
Same here, No output, No log, Nothing, just UI changed... Downgrade to 1.41 works as before.

**Comment by JoeyRxy** [comment]
Same situation. Use 1.4.1 now. Please fix this.

**Comment by sean-mcmanus** [comment]
Sorry about that -- we probably should have shipped the new debuggers in an insiders release instead of putting it directly into 1.5.0.

Is everyone on x86 or x64 machines? Is anyone on an arm64 machine?

**Comment by sean-mcmanus** [comment]
Hmm...I'm not able to repro any bug. Multiple people tests Windows debugging on multiple machine configurations. @WardenGnaw Any ideas how to repro this?

Can someone check if this repros with https://github.com/microsoft/vscode-cpptools/releases/tag/1.5.0-insiders3 or not? That would help confirm if the change is with the most recent Windows vsdbg update or not.

**Comment by sean-mcmanus** [comment]
@WardenGnaw Should we just ship a 1.5.1 with the old Windows x86/x64 debugger or is this easy to fix?

**Comment by zeroxia** [comment]
> Hmm...I'm not able to repro any bug. Multiple people tested Windows debugging on multiple machine configurations. @WardenGnaw Any ideas how to repro this?
> 
> Can someone check if this repros with https://github.com/microsoft/vscode-cpptools/releases/tag/1.5.0-insiders3 or not? That would help confirm if the change is with the most recent Windows vsdbg update or the newer one introduced in 1.5.0-insiders 24 days ago and we just didn't get any reports of this.

I downloaded this 1.5.0-insiders3 .vsix and installed it to test with my previous version 1.4.1 workspace, F5 debugging is running, unlike 1.5.0, nothing happens.

My system is Windows 10 x64. build 19043.

**Comment by sean-mcmanus** [comment]
@zeroxia Are you saying 1.5.0-insiders3 doesn't repro the bug?

**Comment by hostilefork** [comment]
1.5.0-insiders3 does not repro the problem for me.

**Comment by zeroxia** [comment]
> @zeroxia Are you saying 1.5.0-insiders3 doesn't repro the bug?

Correct. I can use 1.5.0-insiders3 just like 1.4.1. Debug session can launch, and break point hit, and I can step code execution.

But with 1.5.0 which auto-updated earlier today, when I hit F5, after cmake-tools reporting build successful, nothing happens, no debug session started at all.

**Comment by Mempler** [comment]
Can confirm `1.5.0-insiders3` & `1.4.1` works perfectly fine while `1.5.0` does not.

**Comment by sean-mcmanus** [comment]
Thanks for the info everyone -- it sounds like @WardenGnaw has found the cause.

**Comment by mcakircali** [comment]
same problem with 1.5.0 update, unable to launch debugger without any error messages. 1.4.1 works...

**Comment by sean-mcmanus** [comment]
Fixed with 1.5.1 (https://github.com/microsoft/vscode-cpptools/releases/tag/1.5.1)

**Comment by Archiatrus** [comment]
> Fixed with 1.5.1 (https://github.com/microsoft/vscode-cpptools/releases/tag/1.5.1)

Unfortunately, not for me. Debugging stops after a split second without any message. Downgrade to 1.4.1 fixes the problem. I did not try 1.5.0.

Bug type: Debugger

    OS and Version: Ubuntu 18.04
    VS Code Version: 1.58.2
    C/C++ Extension Version: 1.5.1

**Comment by Archiatrus** [comment]
Retried it again today with VS Code Version 1.59.0 and still C/C++ Extension Version: 1.5.1 and now debugging works.
